src/libs/terms-of-service-alerts.test.js

```diff
@@ -0,0 +1,82 @@
+import { render } from '@testing-library/react'
+import { act } from 'react-dom/test-utils'
+import { h } from 'react-hyperscript-helpers'
+import Alerts from 'src/components/Alerts'
+import { Ajax } from 'src/libs/ajax'
+import { authStore } from 'src/libs/state'
+import * as TosAlerts from 'src/libs/terms-of-service-alerts'
+
+
+jest.mock('src/libs/ajax')
+jest.mock('src/libs/nav')
+
+jest.mock('react-notifications-component', () => {
+  return {
+    store: {
+      addNotification: jest.fn(),
+      removeNotification: jest.fn()
+    }
+  }
+})
+
+const setupMockAjax = termsOfService => {
+  const getTos = jest.fn().mockReturnValue(Promise.resolve('some text'))
+  const getTermsOfServiceComplianceStatus = jest.fn().mockReturnValue(termsOfService)
+  const getStatus = jest.fn().mockReturnValue({})
+  Ajax.mockImplementation(() => ({
+    Metrics: {
+      captureEvent: jest.fn()
+    },
+    User: {
+      profile: {
+        get: jest.fn().mockReturnValue(Promise.resolve({ keyValuePairs: [] })),
+      },
+      getTos,
+      getTermsOfServiceComplianceStatus,
+      getStatus
+    },
+    FirecloudBucket: {
+      getTosGracePeriodText: jest.fn().mockReturnValue(Promise.resolve('{"text": "Some text"}'))
+    }
+  }))
+}
+
+afterEach(() => {
+  jest.restoreAllMocks()
+  TosAlerts.tosGracePeriodAlertsStore.reset()
+})
+
+const renderAlerts = async termsOfService => {
+  await act(() => Promise.resolve(render(h(Alerts))).finally())
+  setupMockAjax(termsOfService)
+
+  const isSignedIn = true
+  await act(() => Promise.resolve(authStore.update(state => ({ ...state, termsOfService, isSignedIn }))).finally())
```

Comment:
Can we get rid of all the wrapping here? `await act(async () => { authStore.update(...) })` seems to work.